sktime/transformers/panel/minirocket.py

```diff
@@ -0,0 +1,842 @@
+# -*- coding: utf-8 -*-
+__author__ = "Angus Dempster"
+__all__ = ["MiniRocket"]
+
+import numpy as np
+import pandas as pd
+
+from sktime.transformers.base import _PanelToTabularTransformer
+from sktime.utils.validation.panel import check_X
+from numba import njit
+from numba import prange
+from numba import vectorize
+
+
+class MiniRocket(_PanelToTabularTransformer):
+    """MINIROCKET
+
+    MINImally RandOm Convolutional KErnel Transform
+
+    **Univariate**
+
+    Unviariate input only.  Use class MiniRocketMultivariate for multivariate
+    input.
+
+    @article{dempster_etal_2020,
+      author  = {Dempster, Angus and Schmidt, Daniel F and Webb, Geoffrey I},
+      title   = {{MINIROCKET}: A Very Fast (Almost) Deterministic Transform for
+                 Time Series Classification},
+      year    = {2020},
+      journal = {arXiv:2012.08791}
+    }
+
+    Parameters
+    ----------
+    num_features             : int, number of features (default 10,000)
+    max_dilations_per_kernel : int, maximum number of dilations per kernel (default 32)
+    random_state             : int, random seed (optional, default None)
+    """
+
+    _tags = {"univariate-only": True}
+
+    def __init__(
+        self, num_features=10_000, max_dilations_per_kernel=32, random_state=None
+    ):
+        self.num_features = num_features
+        self.max_dilations_per_kernel = max_dilations_per_kernel
+        self.random_state = (
```

Comment:
this should happen in `fit` rather than `init` in case it's changed later via `set_params()`

Reply:
> this should happen in `fit` rather than `init` in case it's changed later via `set_params()`

Sorry, I don't quite understand what you are asking here...  Which part(s) exactly should move from `init` to `fit`?

Reply:
Sorry for not being clearer, the processing of the random state before you assign it to `self.random_state`, I believe that's best done in `fit` via some `check_random_state` function, I now remember the problem you describe with scikit-learn (forgot about that) but we could have our own ROCKET-specific `_check_random_state` function here

Reply:
All good, I'll have a think about a solution here.